### PR TITLE
Add non-reactive circuit breaker implementation of Sentinel

### PR DESCRIPTION
**Pull request by sczyh30**
This PR adds a `spring-cloud-circuitbreaker-sentinel` module as the integration module with [Sentinel](https://github.com/alibaba/Sentinel/). And add non-reactive circuit breaker implementation of `SentinelCircuitBreaker`. The implementation follows the interfaces listed in common module.

The reactive part will be available as soon as the reactor adapter module of Sentinel is released (mid-March).

Please run the test cases to verify it.

Related to #12

### CODE REVIEW

File: spring-cloud-circuitbreaker-sentinel/src/main/java/org/springframework/cloud/circuitbreaker/sentinel/SentinelCircuitBreaker.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @Override
+    public <T> T run(Supplier<T> toRun, Function<Throwable, T> fallback) {
+        Entry entry = null;
+        try {
+            entry = SphU.entry(resourceName, entryType);
+            return toRun.get();
+        } catch (BlockException ex) {
+            return fallback.apply(ex);
```
Comment (ryanjbaxter) [inline spring-cloud-circuitbreaker-sentinel/src/main/java/org/springframework/cloud/circuitbreaker/sentinel/SentinelCircuitBreaker.java]: Is there a reason to distinguish between the different exceptions?

Comment (sczyh30) [reply spring-cloud-circuitbreaker-sentinel/src/main/java/org/springframework/cloud/circuitbreaker/sentinel/SentinelCircuitBreaker.java]: `SphU.entry()` may throw `BlockException` which indicates that the request was rejected (flow control or circuit breaking triggered).  So it should not be counted as the business exception. For other kinds of exceptions, we'll trace the exception count via `Tracer.trace(ex)`.

**Comment by sczyh30** [comment]
Added the starter module for `spring-cloud-circuitbreaker-sentinel`. Plz review :)

File: spring-cloud-circuitbreaker-sentinel/src/main/java/org/springframework/cloud/circuitbreaker/sentinel/SentinelCircuitBreaker.java (hunk 1 above)
Comment (ryanjbaxter) [reply spring-cloud-circuitbreaker-sentinel/src/main/java/org/springframework/cloud/circuitbreaker/sentinel/SentinelCircuitBreaker.java]: Can you add a comment to this effect?

**Comment by ryanjbaxter** [comment]
I recently added documentation to the project, could you add docs for Sentinel as well?

**Comment by sczyh30** [comment]
> I recently added documentation to the project, could you add docs for Sentinel as well?

Sure, I'll add the docs for Sentinel soon.

**Comment by sczyh30** [comment]
@ryanjbaxter document for Sentinel has been added, please check :)

File: README.adoc [hunk 2]
```diff
@@ -30,6 +30,7 @@ breaker implementation that best fits your needs for your app.
 
 * https://github.com/Netflix/Hystrix[Netfix Hystrix]
 * https://github.com/resilience4j/resilience4j[Resilience4J]
+* https://github.com/alibaba/Sentinel[Sentinel]
```
Comment (ryanjbaxter) [inline README.adoc]: No need to make changes to this file, it is automatically generated, see the warning at the top of the file

**Comment by ryanjbaxter** [comment]
I also recently enabled checkstyle rules for the project.  Could you pull in the latest changes from master and fix any checkstyle failures when building the project?

File: README.adoc (hunk 2 above)
Comment (sczyh30) [reply README.adoc]: Thanks for reminding. I've recovered the changes.

**Comment by sczyh30** [comment]
> I also recently enabled checkstyle rules for the project. Could you pull in the latest changes from master and fix any checkstyle failures when building the project?

@ryanjbaxter I've fixed all the checkstyle failures. Please check.

**Pull request merged**